Detectors/TRD/simulation/src/Trap2CRU.cxx

```diff
@@ -530,19 +537,22 @@ int Trap2CRU::buildTrackletRawData(const int trackletindex, const int linkid)
         destroytracklets = true;
         break;
     }
+    //this loop should really be redone but this will work for now.
     trackletcounter++;
+    if (trackletcounter + trackletindex >= mTracklets.size()) {
```

Comment:
Is this a paranoia check or is this expected to happen? If it is a paranoia check maybe one could add a log message? Similar for the link checks for digits and tracklets